### PR TITLE
Add functions to calculate squad/unit infiltration

### PR DESCRIPTION
**Pull request by cannonfodder1**
Added some things that might be useful when we start fully connecting the squad select to the CA system.

### CODE REVIEW

**Comment by Xymanek** [comment]
I don't think `X2InfiltrationModTemplateManager` Is the place for these things.... Maybe make a new class? `InfiltrationDurationCalculator` or something?

**Comment by robojumper** [comment]
I don't have a problem with the Template Manager. But `InfiltrationDurationCalculator` strikes me as a bad class name -- it's not the class as a whole that's doing the calculation, it's a function -- and we'll probably add some more functions to this. Maybe `InfiltrationUtils`?

**Comment by Xymanek** [comment]
I think we should give player as much info on infiltration duration as possible - similar to shot hit/crit/damage breakdown. So in this case it would be duration from units, gear, etc. That will probably induce much more code - that's why i suggested separating this stuff into its own class

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc [hunk 1]
```diff
@@ -0,0 +1,80 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  NotSoLoneWolf
+//  PURPOSE: This class contains a variety of tools
+//           for this mod and others to interact with
+//           the infiltration template system.
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class X2Helper_Infiltration extends Helpers config(Infiltration);
```
Comment (robojumper) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: Why `Helpers`? Extend `Object` until you have a good reason not to.

Comment (Xymanek) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: Yeah, I looked at `Helpers`, there is nothing we really need there (and if we do, then we just use `class'Helpers'.static.Whatever()`)

File: CovertInfiltration/Src/CovertInfiltration/Classes/UIAvengerHUD_CAButton.uc [hunk 2]
```diff
@@ -17,15 +17,13 @@ event OnInit(UIScreen Screen)
 	local UIStrategyMap StrategyMap;
 
 	StrategyMap = UIStrategyMap(Screen);
-	if (StrategyMap == none) return;
+	if(StrategyMap == none || `HQPRES.StrategyMap2D == none || `HQPRES.StrategyMap2D.m_eUIState == eSMS_Flight) return;
```
Comment (robojumper) [inline CovertInfiltration/Src/CovertInfiltration/Classes/UIAvengerHUD_CAButton.uc]: I don't understand this change. Skip spawning any button *at all* when the map is currently in flight mode? When can this happen?

Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/UIAvengerHUD_CAButton.uc]: Oh yeah, that was a failed attempt at fixing the "CA button shows up when avenger is in flight" issue. I forgot to remove it and it got pushed to this branch by accident.

**Comment by cannonfodder1** [comment]
Not sure why they didn't get resolved because I did add checks for `none`. Few more QOL changes and I'm merging

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		{
+			TotalInfil = InfilMgr.GetInfilTemplateFromItem(InventoryItem.GetMyTemplateName()).InfilModifier;
+			ItemCategory = InventoryItem.GetMyTemplate().ItemCat;
+
+			foreach CurrentInventory(MultiplierItem)
+			{
+				if(MultiplierItem != none)
+				{
+					MultCategory = InfilMgr.GetInfilTemplateFromItem(MultiplierItem.GetMyTemplateName()).MultCategory;
+					Multiplier = InfilMgr.GetInfilTemplateFromItem(MultiplierItem.GetMyTemplateName()).InfilMultiplier;
```
Comment (robojumper) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: I'm not seeing any `none` checks though...

Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: It literally says it in the code block you highlighted though?

Comment (robojumper) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: `InfilMgr.GetInfilTemplateFromItem(MultiplierItem.GetMyTemplateName())` can be none, then `….MultCategory` throws an accessed none warning. Same for the other instances of `GetInfilTemplateFromItem`.

Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: Alright, I've changed the none checks to look like this:
`(InfilMgr.GetInfilTemplateFromItem(InventoryItem.GetMyTemplateName()) != none)`
For both InventoryItem and MultiplierItem.

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationMod.uc [hunk 4]
```diff
@@ -31,35 +19,35 @@ enum EInfiltrationWeight
...
+var config int WeaponInfiltration[EInfiltrationWeight.EnumCount]<BoundEnum=EInfiltrationWeight>;
+
+enum EDeterrenceLevel
+{
+	eDeterrenceLevel_Unnoticed,
+	eDeterrenceLevel_Trivial,
+	eDeterrenceLevel_Concerning,
+	eDeterrenceLevel_Threatening,
+	eDeterrenceLevel_Intimidating
+};
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationMod.uc]: Please put enums together at the top

**Comment by robojumper** [comment]
Rebased.

**Comment by Xymanek** [comment]
@cannonfodder1 when you see this, delete `squad-infil-calc` branch in the web ui and locally (since it's merged now)

**Comment by Xymanek** [comment]
[Link duration of CAs to soldiers and gear (PARTIAL)](https://trello.com/c/ADun5nvq/28-link-duration-of-cas-to-soldiers-and-gear-partial)

**Comment by Xymanek** [comment]
[Link the risks to soldiers and gear](https://trello.com/c/yGCs9ea5/38-link-the-risks-to-soldiers-and-gear)

**Pull request merged**